design/bytecode/README.md

```diff
@@ -1,82 +1,117 @@
+| purpose | conjunction | not |  operation | RESERVED |
 
-```rust
-// Load a datum from the input data table into the next free register
-LD_INPUT = 0
-// Load a datum from the user data table into the next free register
-LD_USER = 1
-/* Conjunctions */
-// Compute a logical and between A and the next comparator OpCode
-AND = 2
-// Compute an inclusive or between A and the next comparator OpCode
-OR = 3
-// Compute an exclusive or between A and the next comparator OpCode
-XOR = 4
-/* Comparators */
-// i == j
-EQ = 5
-// i > j
-GT = 6
-// i >= j
-GTE = 7
-// Whether data[i] is included in the set at data[j]
-IN = 8
-// i < j
-LT = 9
-// i <= j
-LTE = 10
-```
+  - `bit(7)` determines whether the opcode is a conjunction or a comparator
+    ```rust
+      // OpCode represents a comparator
+      COMP = 0
+      // OpCode represents a conjunction
+      CONJ = 1
+    ```
+  - `bit(6)` determines whether a NOT is applied to the conjunction/comparator
+    ```rust
+      // No logical inversion
+      NORMAL = 0
+      // Invert the logic of the output
+      NOT = 1
+    ```
+  - `bits(5..2)` (4 bits):
+    - for comparators (`bit(7) == 0`):
+      | bits    |    5 |      4 - 2 |
+      |:--------|-----:|-----------:|
+      | purpose | load | comparator |
+      - `bit(5)` determines the LOADs to compare:
+        ```rust
+        // Compare from input to datatable entries
+        LOAD_INPUT_VS_USER = 0;
+        // Compare from input to input entries
+        LOAD_INPUT_VS_INPUT = 1;
+        ```
+      - `bits(4..2)` determines the comparator operation
+        ```rust
+        // i == j
+        EQ = 0
+        // i > j
+        GT = 1
+        // i >= j
+        GTE = 2
+        // Whether data[i] is included in the set at data[j]
+        IN = 3
+        ```
+        *Note: `LT` and `LTE` are achieved by using `bit(6)`, the `NOT` operator.*
+    - for conjunctions (`bit(7) == 1`):
+      ```rust
+      // Compute a logical and between A and the next comparator OpCode
+      AND = 0
+      // Compute an inclusive or between A and the next comparator OpCode
+      OR = 1
+      // Compute an exclusive or between A and the next comparator OpCode
+      XOR = 2
+      ```
+  - `bits(1..0)` are RESERVED
+
+## Index Codes
+
+A Pact may have up to 16 input arguments and up to 16 entries in a user data table.
+
+This limitation allows comparator indices to be encoded in a single byte:
+
+| bits    |      7 - 4 |     3 - 0 |
+|:--------|:----------:|:---------:|
+| purpose |  LHS index | RHS index |
 
 ## Example Syntax
 
 A series of independent clauses ("implicit and")
 ```pact
-EQ LD_INPUT 0 LD_USER 1
-GT LD_INPUT 2 LD_USER 2
-LT LD_INPUT 1 LD_USER 0 
+(COMP + LOAD_INPUT_VS_USER + EQ) ((0 << 4) + 1)            # INPUT(0) == USER(1)    | 0x00, 0x01
+(COMP + LOAD_INPUT_VS_USER + GT) ((0 << 4) + 1)            # INPUT(0) >  USER(1)    | 0x04, 0x01
+(COMP + NOT + LOAD_INPUT_VS_USER + GTE) ((0 << 4) + 1)     # INPUT(0) <  USER(1)    | 0x48, 0x01
 ```
```

Comment:
Optional: I'm not sure what "((0 << 4) + 1) " mean. Perhaps a Binary expression here is more helpful?
i.e. 0000 0000, 0000 0001
0000,0100, 0000 0001

Reply:
Update: I see the second part indicate the Load position. Might wanna use a non-zero example to make it more clear

Reply:
I changed the examples to be a little clearer